src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2907,14 +2907,15 @@ IsPartitionColumnRecursive(Expr *columnExpression, Query *query)
 	List *rangetableList = query->rtable;
 	Index rangeTableEntryIndex = 0;
 	RangeTblEntry *rangeTableEntry = NULL;
+	Expr *strippedColumnExpression = strip_implicit_coercions(columnExpression);
```

Comment:
@jasonmp85  @aamederen  I get compile warnings with this line on MacOS with the following gcc:
```
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin15.0.0
Thread model: posix
```
 
The warning:
```
planner/multi_logical_optimizer.c:2911:60: warning: incompatible pointer types passing 'Expr *' (aka 'struct Expr *') to parameter of type 'Node *'
      (aka 'struct Node *') [-Wincompatible-pointer-types]
        Expr *strippedColumnExpression = strip_implicit_coercions(columnExpression);
                                                                  ^~~~~~~~~~~~~~~~
/usr/local/pgsql/include/server/nodes/nodeFuncs.h:33:45: note: passing argument to parameter 'node' here
extern Node *strip_implicit_coercions(Node *node);
                                            ^
planner/multi_logical_optimizer.c:2911:8: warning: incompatible pointer types initializing 'Expr *' (aka 'struct Expr *') with an expression of type
      'Node *' (aka 'struct Node *') [-Wincompatible-pointer-types]
        Expr *strippedColumnExpression = strip_implicit_coercions(columnExpression);
```